src/x/xsystem.c

```diff
@@ -354,6 +360,16 @@ static bool xglx_get_cursor_position(int *ret_x, int *ret_y)
 static bool xglx_inhibit_screensaver(bool inhibit)
 {
    ALLEGRO_SYSTEM_XGLX *system = (void *)al_get_system_driver();
+   int temp, version_min, version_max;
+
+#ifdef ALLEGRO_XWINDOWS_WITH_XSCREENSAVER
+   if (!XScreenSaverQueryExtension(system->x11display, &temp, &temp) ||
+      !XScreenSaverQueryVersion(system->x11display, &version_max, &version_min) ||
+      version_max < 1 || (version_max == 1 && version_min < 1)) {
+      return false;
```

Comment:
Is this already taken care of in `_al_xwin_background_thread`? (xevents.c:227)

Reply:
Elias means something like this: https://gist.github.com/SiegeLord/7176a2ea807df9a145d4872b5ccb0322 I.e. instead of `#ifndef` in `xevents.c`, we use a boolean. The idea is if someone compiles Allegro with this query enabled, but then distributes the binary to a system without that extension, we want to make a runtime check for this feature.

Reply:
Brilliant - thanks for the pointers. Sorry for my lack of knowledge! I've applied something very close to that diff, just a couple of rearrangements.